utility/log.py

```diff
@@ -1,66 +1,54 @@
-# -*- coding: utf-8 -*-
-"""
-Implements CephCI logging interface.
-
-In this module, we implement a singleton LOG object that can be used by all components
-of CephCI. It supports the below logging methods
-
-    - error
-    - warning
-    - info
-    - debug
-
-Along with the above, it provides support for pushing events to
-
-    - local file
-    - logstash server
-
-Initial Log format will be 'datetime - level - message'
-later updating log format with 'datetime - level -filename:line_number - message'
-"""
-import inspect
 import logging
 import logging.handlers
 import os
-from copy import deepcopy
-from typing import Any, Dict
+from typing import Dict
 
 from .config import TestMetaData
 
-LOG_FORMAT = "%(asctime)s (%(name)s) [%(levelname)s] - %(message)s"
+LOG_FORMAT = "%(asctime)s (%(name)s) - %(module)s:%(lineno)d - %(funcName)s - [%(levelname)s] - %(message)s"
+
 magna_server = "http://magna002.ceph.redhat.com"
 magna_url = f"{magna_server}/cephci-jenkins/"
 
 
-class LoggerInitializationException:
+class LoggerInitializationException(Exception):
+    """Exception raised for logger initialization errors."""
+
     pass
 
 
-class Log:
+class Log(logging.Logger):
     """CephCI Logger object to help streamline logging."""
 
     def __init__(self, name=None) -> None:
-        """Initializes the logging mechanism based on the inputs provided."""
-        self._logger = logging.getLogger("cephci")
+        """
+        Initializes the logging mechanism.
+        Args:
+            name (str): Logger name (module name or other identifier).
+        """
+        super().__init__(name)
         logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+        self._logger = logging.getLogger(name)
 
+        # Set logger name
         if name:
-            self._logger.name = f"cephci.{name}"
+            self.name = f"cephci.{name}"
 
-        self._log_level = self._logger.getEffectiveLevel()
+        # Additional attributes
+        self._log_level = self.getEffectiveLevel()
         self._log_dir = None
-        self._log_errors = []
         self.log_format = LOG_FORMAT
+        self._log_errors = []
+        self.info = self._logger.info
+        self.debug = self._logger.debug
+        self.warning = self._logger.warning
+        self.error = self._logger.error
```

Comment:
We have a method `error` and an attribute `error` likewize `self._logger.error` is a method. This is a bit confusing.

Since we are adding messages as part of `error` method, I believe we shouldn't have this attribute.

Reply:
We have attribute `self._log_errors = []` which we are using to log all the errors collected. This list is used in run.py in Exception block to display the errors